sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3191,8 +3198,11 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
   case PI_DEVICE_INFO_MAX_MEM_BANDWIDTH:
     // currently not supported in level zero runtime
     return PI_ERROR_INVALID_VALUE;
-  case PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16:
-    return PI_ERROR_INVALID_VALUE;
+  case PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16: {
+    // bfloat16 is implemented in hardware or emulated, so it is always
+    // supported.
+    return ReturnValue(bool{true});
```

Comment:
I suppose we should make the same change to pi_cuda here: https://github.com/intel/llvm/blob/9f89247295495b15f10176fbb43c2fe756077505/sycl/plugins/cuda/pi_cuda.cpp#L1305

Reply:
Is this code used to determine the return value of `device::has(aspect::ext_oneapi_bfloat16)`?  If so, then `true` is not the right implementation.  We decided that this aspect now tells whether the bfloat16 math functions can be called on the device, and I believe these math functions are only allowed on devices that have native support, correct?

If that is the case, then the implementations of `PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16` are probably correct on the OpenCL and CUDA backends, and the implementation is wrong for Level Zero.

Reply:
We now want the aspect bfloat16 to mean support for bfloat16 math functions.
The CUDA plugin can stay the way it is because it supports the math functions.
The OpenCL and L0 the plugins will report "unsupported".